vignettes/query.Rmd

```diff
@@ -0,0 +1,212 @@
+---
+title: "Outpack query DSL"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Outpack query DSL}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+Outpack includes a query DSL (domain specific language), extending the one used by orderly (see [`orderly1::orderly_search()`](https://www.vaccineimpact.org/orderly/reference/orderly_search.html)).
+
+Queries are used in identifying ids to pull in as dependencies, so rather than providing an identifier, you might want to depend on
+
+* the most recent copy of a packet with a particular name
+* ...produced in some date range
+* ...with some particular set of parameter values
+* ...that produced a particular file
+* ...that was depended on by some other packet
+
+Not all of this is supported as of the current version of `orderly2`, and *using* the queries seamlessly is also not supported!
+
+## Structure of queries
+
+The most simple query is
+
+```r
+latest()
+```
+
+which finds the most recent packet; this is unlikely to be very useful without scoping - see below.
```

Comment:
I've ended up a little confused in this doc - so returning to the top after reading it...

1. I think I'd find it easier if we started with a "useful" query. This one, if I understand it right, has us depending on the latest report in the archive, regardless of what the report was - and I agree that's unlikely to be useful without scoping (it probably doesn't make sense really)

2. So adding the scope is familiar if you're used to orderly-1, but then we have a "Warning we may remove this" at the top of the Scoping section... 

So I think maybe an approach is to decide whether or not we want scoping to be kept. If it does, then describe it up here as a helping hand for orderly-1 veterans, and say the new `name` parameter is the new and shiny way of doing it. Or otherwise, kill scoping off properly and don't reference it here; just go straight in with a useful-ish (and maybe more concrete/eg covid) example that uses both latest, and a named report.

Reply:
Agreed, once the querying settles down a bit we should probably make this more accessible (volunteers welcome, but this will be a bit of a job I suspect, with lots of concrete examples)